pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/persistence/Serializers.kt

```diff
@@ -0,0 +1,282 @@
+            val y = buffer.getInt()
+            val z = buffer.getInt()
+            return BlockPosition(Bukkit.getServer().getWorld(UUID(mostSignificantBits, leastSignificantBits)), x, y, z)
+        }
+        else{
+            val buffer = ByteBuffer.wrap(primitive)
+            val x = buffer.getInt()
+            val y = buffer.getInt()
+            val z = buffer.getInt()
+            return BlockPosition(null, x, y, z)
+        }
+    }
+
+    override fun toPrimitive(complex: BlockPosition, context: PersistentDataAdapterContext): ByteArray {
+        if(complex.world != null){
+            val buffer = ByteBuffer.allocate(2 * Long.SIZE_BYTES + 3 * Int.SIZE_BYTES)
+            buffer.putLong(complex.world!!.uid.mostSignificantBits)
+            buffer.putLong(complex.world!!.uid.leastSignificantBits)
+            buffer.putInt(complex.x)
+            buffer.putInt(complex.y)
+            buffer.putInt(complex.z)
+            return buffer.array()
+        }
+        else{
+            val buffer = ByteBuffer.allocate(3 * Int.SIZE_BYTES)
+            buffer.putInt(complex.x)
+            buffer.putInt(complex.y)
+            buffer.putInt(complex.z)
+            return buffer.array()
+        }
+    }
+}
+
+class ChunkPositionPersistentDataType : PersistentDataType<ByteArray, ChunkPosition> {
+    override fun getPrimitiveType(): Class<ByteArray>
+        = ByteArray::class.java
+
+    override fun getComplexType(): Class<ChunkPosition>
+        = ChunkPosition::class.java
+
+    override fun fromPrimitive(primitive: ByteArray, context: PersistentDataAdapterContext): ChunkPosition {
+        if(primitive.size > 2 * Int.SIZE_BYTES){
+            val buffer = ByteBuffer.wrap(primitive)
+            val mostSignificantBits = buffer.getLong()
+            val leastSignificantBits = buffer.getLong()
+            val x = buffer.getInt()
+            val z = buffer.getInt()
+            return ChunkPosition(Bukkit.getServer().getWorld(UUID(mostSignificantBits, leastSignificantBits)), x, z)
+        }
+        else{
+            val buffer = ByteBuffer.wrap(primitive)
+            val x = buffer.getInt()
+            val z = buffer.getInt()
+            return ChunkPosition(null, x, z)
+        }
+    }
+
+    override fun toPrimitive(complex: ChunkPosition, context: PersistentDataAdapterContext): ByteArray {
+        if(complex.world != null){
+            val buffer = ByteBuffer.allocate(2 * Long.SIZE_BYTES + 2 * Int.SIZE_BYTES)
+            buffer.putLong(complex.world!!.uid.mostSignificantBits)
+            buffer.putLong(complex.world!!.uid.leastSignificantBits)
+            buffer.putInt(complex.x)
+            buffer.putInt(complex.z)
+            complex.asLong
+            return buffer.array()
+        }
+        else{
+            val buffer = ByteBuffer.allocate(2 * Int.SIZE_BYTES)
+            buffer.putInt(complex.x)
+            buffer.putInt(complex.z)
+            return buffer.array()
+        }
```

Comment:
```suggestion
        val buffer = ByteBuffer.allocate(2 * Long.SIZE_BYTES + 2 * Int.SIZE_BYTES)
        val world = complex.world
        if (world != null) {
            buffer.putLong(world.uid.mostSignificantBits)
            buffer.putLong(world.uid.leastSignificantBits)
        }
        buffer.putInt(complex.x)
        buffer.putInt(complex.z)
        return buffer.array()
```
This can be massively simplified

Reply:
Yeah I suggested using a byte to indicate whether the world is present in mine but this would also work. Up to you ohm

Reply:
eh I'd rather not, this is the part of the code that kinda needs to get hyperoptimized